src/layers/mpls.rs

```diff
@@ -0,0 +1,72 @@
+//! Multi Label Protocol Switching
+
+use std::convert::TryInto;
+
+use serde::Serialize;
+
+use crate::errors::Error;
+use crate::layers::ethernet;
+use crate::types::{ETHERTYPE_MPLS_MULTICAST, ETHERTYPE_MPLS_UNICAST};
+use crate::Layer;
+
+use super::ipv4::IPv4;
+
+/// Default Header Length for MPLS Packets
+pub const MPLS_HDR_LEN: usize = 4_usize;
+
+// Register Ourselves to the Ethernet layer, as this is a 2.5 layer protocol
+pub(crate) fn register_defaults() -> Result<(), Error> {
+    ethernet::register_ethertype(ETHERTYPE_MPLS_UNICAST, MPLS::creator)?;
+    ethernet::register_ethertype(ETHERTYPE_MPLS_MULTICAST, MPLS::creator)
+}
+
+#[derive(Debug, Default, Serialize)]
+pub struct MPLS {
+    //Make built-in types
+    #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u32")]
+    label: u32, //This is only 20 bits
+    #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u8")]
+    exp: u8, //This is only 3 bits
+    #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u8")]
+    bos: u8, //This is only 1 bit
```

Comment:
Make it a `bool` more semantic.

Reply:
The `bool` is getting printed as `true` or `false`, is that okay?